packages/mermaid/src/diagram-api/frontmatter.ts

```diff
@@ -0,0 +1,39 @@
+// The "* as yaml" part is necessary for tree-shaking
+import * as yaml from 'js-yaml';
+
+// Match Jekyll-style front matter blocks (https://jekyllrb.com/docs/front-matter/).
+// Based on regex used by Jekyll: https://github.com/jekyll/jekyll/blob/6dd3cc21c40b98054851846425af06c64f9fb466/lib/jekyll/document.rb#L10
+// Note that JS doesn't support the "\A" anchor, which means we can't use
+// multiline mode.
+// Relevant YAML spec: https://yaml.org/spec/1.2.2/#914-explicit-documents
+export const frontMatterRegex = /^(?:\s*---\s*[\r\n])(.*?)(?:[\r\n]\s*---\s*[\r\n]+)/s;
+
+type FrontMatterMetadata = {
+  title?: string;
+};
+
+/**
+ * Extract and parse frontmatter from text, if present, and sets appropriate
+ * properties in the provided db.
+ * @param text -
+ * @param db -
+ * @returns text with frontmatter stripped out
+ */
+export function extractFrontMatter(text: string, db: any): string {
```

Comment:
We should probably give `db` a type, even if it is a very basic one.

Could we do something like:

```ts
/**
 * Generic Diagram DB that may apply to any diagram type.
 */
interface DiagramDb {
    setDiagramTitle?: (title: string) => void
};
```

Ideally, we'd use classes/inheritance for the databases, but that will be a big refactor for the future.

Reply:
Done in 3316aa5f4f8c93f3410ca74c2fa5390087f26ef9